api/base/filters.py

```diff
@@ -44,36 +78,19 @@ class ODMFilterMixin(object):
 
     # TODO Handle simple and complex non-standard fields
 
-    TRUTHY = set(['true', 'True', 1, '1'])
-    FALSY = set(['false', 'False', 0, '0'])
-    DEFAULT_OPERATOR = 'eq'
-
     # For the field_comparison_operators, instances can be a class or a tuple of classes
-    field_comparison_operators = [
-        {
-            'field_type': ser.CharField,
-            'comparison_operator': 'icontains'
-        },
-        {
-            'field_type': ser.ListField,
-            'comparison_operator': 'in'
-        }
-    ]
+    field_comparison_operators = {
+        ser.CharField: 'icontains',
+        ser.ListField: 'in',
+    }
 
     def get_comparison_operator(self, key):
-
-        for operator in self.field_comparison_operators:
-            if isinstance(self.serializer_class._declared_fields[key], operator['field_type']):
-                return operator['comparison_operator']
+        for field_type, comparison_operator in self.field_comparison_operators.items():
```

Comment:
I was thinking in terms of using `type()` rather than `isinstance`. 

```python
field = self.serializer_class._declared_fields[key]
operator = self.field_comparison_operators.get(type(field), self.DEFAULT_OPERATOR)
```

Indeed, using direct type checking means that subclasses would need to be added to `field_comparison_operators`, though I expect that to be a rare case.